### ISSUE TITLE
_TARGET_OS does not point to macosx when specifying --os macosx

### ISSUE THREAD

**Issue by SanderMertens** [premake/premake-core#977]
I have a premake file in which I want to output to different directory based on the tool and operating system that I'm generating for. In my premake file I have the following line:
```lua
location (_ACTION .. "-" .. _TARGET_OS)
```
When I do `premake5 gmake` it outputs to `gmake-linux`. However, if I do `premake5 gmake --os macosx` it still outputs to `gmake-linux`.

I also tried setting the location under different configurations, but couldn't get that to work either.

**Comment by samsinsane** [comment]
[`os.target()`](https://github.com/premake/premake-core/blob/08c768d20e931953b80907381cc0b0bd64286892/src/base/os.lua#L186-L188) will get you what you're after. To my understanding, the `os` option isn't perfect, so please bug any issues you encounter when using it and `os.target()`.

**Comment by SanderMertens** [comment]
Will give this a go. Thanks!

**Comment by starkos** [comment]
Probably worth leaving this one open as a bug.

**Comment by samsinsane** [comment]
> Probably worth leaving this one open as a bug.

Just so I'm on the same page, is the bug that `--os` *doesn't* set `_TARGET_OS`?

**Comment by starkos** [comment]
That's right.

**Comment by neico** [comment]
[This](https://github.com/premake/premake-core/blob/5a0b07cd45aeb0d3a90a28a34744c2f610655abc/src/base/os.lua#L187) is probably why, when it was refactored apparently only the functions got the attention

`_TARGET_OS` is only set in premake.c via `PLATFORM_STRING`, but not trough the commandline

And the tests didn't catch this because they override `_TARGET_OS` and `_OPTIONS.os` before running the tests, always resulting in correct behaviour during testing.

**Comment by liampace** [comment]
I am running into this same issue with my project. The [documentation](https://premake.github.io/docs/globals/_TARGET_OS/) seems to indicate that the intended behavior for the `--os` flag is supposed to override `_TARGET_OS` so this is an error in either premake itself or it's documentation

**Comment by nickclark2016** [comment]
Correct, we should either adjust the documentation or adjust the behavior of `--os` and `_TARGET_OS`. I would recommend using `target.os()` for now.

**Comment by liampace** [comment]
I tested `os.target` and it seems to have the correct behavior which is funny in a bizarre sense as the global variable named after it does not :/

I'm using a temporary fix where I have a lua module who's sole purpose is to set `_TARGET_OS` to `os.target()` lol
